### DISCUSSION TITLE
Convex and mesh geometry not getting overlap result as expected

### DISCUSSION THREAD

**Discussion by yananZZZZ** [NVIDIA-Omniverse/PhysX#145] [General]
Hi,

I tried to use scene query to simulate an entry gate whatever collision object enter and exit my volume I would have some action. This concept works fine for box, capsule. But NOT for convex and mesh. 
I check the manual and for convex shape the scene query is supported and the API I used is "PxSceneQueryExt::overlapMultiple". Is there any possible reason that fail on these two type of geometries? The geometry and the position of geometry is corrected I have them double checked.

**Comment by yananZZZZ** [comment]
One more interesting thing is that I create a collision shape with the same geometry the collision works just fine. I wonder if there is some tolerance or different scale between scene query and collision?

**Comment by preist-nvidia** [comment]
You want to check an overlap between a static triangle mesh and a convex, is that correct?

**Reply by yananZZZZ** [reply]
Hi,
What's I want is simple, I just want to use convex or mesh geometry to do the overlap test among any shape in the scene. The shape type could be any, box, sphere, mesh or even custom geometry. Is that supported?  
What's confused me is that the same convex geometry, when I create a shape and associate it with a static actor at the same position, any other shape can collide with it however, my overlap function returns empty hit.   
This is my query code, maybe i miss some flag or something?
    _PxQueryFilterData filterData;
    filterData.data = filter;
    filterData.flags = (PxQueryFlag::PxQueryFlag::eSTATIC | PxQueryFlag::eDYNAMIC | PxQueryFlag::eNO_BLOCK | PxQueryFlag::ePOSTFILTER);
    m_hitCount = PxSceneQueryExt::overlapMultiple(*m_pScene, geometry, pose, m_hitBuffer, m_bufferSize, filterData, filterCall);_

Thank you.

**Reply by preist-nvidia** [reply]
Did you compare your code to our Snippet using overlapMultiple (press F8)?: https://github.com/NVIDIA-Omniverse/PhysX/tree/release/104.2/physx/snippets/snippetquerysystemallqueries

Or could you adapt this snippet to reproduce the misses that you are seeing?
